docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,478 @@
+CommandResponseTopicPrefix = "edgex/command/response/"  # publishing outbound command responses; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+!!! Note
+        Most of the device service configuration is existing based on its need to already communicate with the message bus for publishing events/readings.  The last two lines are added to allow device services to subscribe and publish command messages from/to the message bus.
+
+## Questions
+
+- Do we need separate topics for all the devices or would one on the device service suffice?
+    
+    - Ans:  we have defined the deviceName in the parameterized topic, so one topic should be sufficient for Device Service- edgex/command/request/<device-name>...
+
+- Would clients (non EdgeX services and applications) want to get a list of available commands via message (instead of calling REST)?
+
+    - Ans:  this is a valid question and could be provided via later additions to the command service (or other service like metadata) in the future.  It does not have to be tackled immediately.  
+
+- Dynamic configuration of the message subscription is not a user friendly operation today (requiring configuration changes).
+
+    - Ans:  In the future, we might want to think about creating additional APIs for Adding/Updating/Deleting/Query the external subscription (and store them to the RedisDB).
+
+- Is it acceptable for more than one response to be published by the device service on the same correlation ID? Eg, send back "Acknowledged", then "Scheduled", then "Starting", then "Done" statuses?
+    
+    - Ans:  No, the correlation id has a life span to/from the initial requester to the response back to the requester.
+
+- Would it make sense to echo the command name into the response, as a reality check?
+    - Ans: solved via topic naming.  Also, per @lenny-intel: "not needed as we don't do this in the HTTP response. The response topic doesn't need the extra path info. The request ID or correlation ID is all that is needed to match the response to the request. No need to make it more complex."
+
+- Would sending/receiving binary data (e.g. CBOR) be supported in this north-south message implementation?
+    
+    - Ans:  today, command service and device services support CBOR get operations but not set (C SDK suppports both).  [Suggest getting feature parity](https://github.com/edgexfoundry/device-sdk-go/issues/488) in place between the SDKs before exploring CBOR support messaging binary/CBOR payloads.
+    - Ans update per Monthly Architect's meeting of 2/28/22: support get with CBOR payload.
+
+- Use of the message bus communications (by the non-EdgeX 3rd party service or application) would bypass the API Gateway.
+
+    - Ans:  not an issue since the command service is serving as external to internal message bus broker.
+
+- Note a number of open questions in the Message Structure section that still need to be addressed.
```

Comment:
This ^^

Reply:
addressed in monthly architect's meeting of 3/28/22